Automattic/wp-calypso
# Domains: handle pending registration at registry

## Proposed Changes

.com.br domain registration are asynchronous. It can take up to 3 days for a domain to be registered.

This PR shows a message for domains with a pending registration and disables all the setting cards (with the exception of `Details`) and operations not allowed (transfers and email creation).

This PR depends on D129922-code

## Testing Instructions

Apply the and verify that the domain setting page is rendered as in the screenshot below for a .com.br with a pending registration (if you don't have a domain in this status hack the backed to force `pending_registration_at_registry` property. 

![pending-registration](https://github.com/Automattic/wp-calypso/assets/2797601/cebfd6ba-8abe-4c1e-8e02-1c9bde68435b)


## Pre-merge Checklist
